File: scratch/inspect_images.py
import os
from PIL import Image

assets_dir = r"c:\Users\ravip\AndroidStudioProjects\PoliceMobileDirectory\assets"
print(f"Inspecting assets in: {assets_dir}")

if not os.path.exists(assets_dir):
    print("Assets directory does not exist!")
    exit(1)

files = os.listdir(assets_dir)
print(f"Total files: {len(files)}")

for i, f in enumerate(files):
    if f.lower().endswith(('.png', '.jpg', '.jpeg')):
        full_path = os.path.join(assets_dir, f)
        try:
            with Image.open(full_path) as img:
                print(f"[{i}] {f}")
                print(f"    Size: {os.path.getsize(full_path)} bytes")
                print(f"    Format: {img.format}, Mode: {img.mode}, Dimensions: {img.size}")
        except Exception as e:
            print(f"[{i}] {f} - Error: {e}")

File: scratch/inspect_images.js
const fs = require('fs');
const path = require('path');

const assetsDir = "c:\\Users\\ravip\\AndroidStudioProjects\\PoliceMobileDirectory\\assets";
console.log(`Inspecting headers in: ${assetsDir}`);

const files = fs.readdirSync(assetsDir);
files.forEach((f, idx) => {
    if (f.toLowerCase().endsWith('.png')) {
        const fullPath = path.join(assetsDir, f);
        try {
            const fd = fs.openSync(fullPath, 'r');
            const buffer = Buffer.alloc(16);
            fs.readSync(fd, buffer, 0, 16, 0);
            fs.closeSync(fd);
            
            const hex = buffer.toString('hex').toUpperCase();
            console.log(`[${idx}] ${f}`);
            console.log(`    Hex: ${hex}`);
            // Check for WebP
            if (hex.startsWith("52494646") && hex.includes("57454250", 16)) { // Wait, 57454250 is WEBP
                console.log(`    Type: WebP`);
            } else if (hex.startsWith("89504E47")) {
                console.log(`    Type: PNG`);
            } else if (hex.startsWith("FFD8FF")) {
                console.log(`    Type: JPEG`);
            } else {
                console.log(`    Type: Unknown`);
            }
        } catch (e) {
            console.log(`[${idx}] ${f} - Error: ${e.message}`);
        }
    }
});
